source/Templates/Sashimi.Template.Wrangler/Program.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Sashimi.Template.Wrangler
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            string packageVersion = null;
+            string installPath = null;
+
+            if (args.Length < 2)
+            {
+                throw new Exception($"Invalid operation, Usage {typeof(Program).Namespace} <packageVersion> <installPath>");
+            }
+
+            if (args.Length == 2)
+            {
+                packageVersion = args[0];
+                installPath = args[1];
+            }
+
+            var solutionPath = Path.GetFullPath(Path.Combine(installPath, "source"));
+
+            var csProjFiles = Directory.EnumerateFiles(solutionPath, "*.csproj", SearchOption.AllDirectories);
+            var packagesToAdd = new Dictionary<string, string>();
+
+            ProcessUtils.RunDotNetCommand("new", solutionPath, "sln -n Sashimi.NamingIsHard");
+
+            foreach (var projFile in csProjFiles)
+            {
+                var sb = new StringBuilder();
+
+                foreach (var line in File.ReadLines(projFile))
```

Comment:
Would it be possible to add the projects to the solution first (in an invalid state), then fix up the references? This way you could move Line 60 into the the loop on Line 34

Reply:
I had a look and I don't think we can do it, because the first pass it strips out project references. Happy for you to have a go?